src/providers/MeetingProvider/MeetingManager.ts

```diff
@@ -325,23 +323,54 @@ export class MeetingManager implements AudioVideoObserver {
 
   audioVideoDidStop = (sessionStatus: MeetingSessionStatus) => {
     const sessionStatusCode = sessionStatus.statusCode();
-    if (sessionStatusCode === MeetingSessionStatusCode.AudioCallEnded) {
-      console.log('[MeetingManager audioVideoDidStop] Meeting ended for all');
-      this.meetingStatus = MeetingStatus.Ended;
-      this.publishMeetingStatus();
-    } else if (sessionStatusCode === MeetingSessionStatusCode.AudioJoinedFromAnotherDevice) {
-      console.log('[MeetingManager audioVideoDidStop] Meeting joined from another device');
-      this.meetingStatus = MeetingStatus.JoinedFromAnotherDevice;
-      this.publishMeetingStatus();
-    } else {
-      console.log(`[MeetingManager audioVideoDidStop] session stopped with code ${sessionStatusCode}`);
+    switch (sessionStatusCode) {
+      case MeetingSessionStatusCode.AudioCallEnded: 
+        console.log('[MeetingManager audioVideoDidStop] Meeting ended for all');
+        this.meetingStatus = MeetingStatus.Ended;
+        this.publishMeetingStatus();
+        this.leave();
+        break;
+      case MeetingSessionStatusCode.Left:
+        console.log('[MeetingManager audioVideoDidStop] Left the meeting');
+        this.meetingStatus = MeetingStatus.Left;
+        this.publishMeetingStatus();
+        // No need to call leave() here, since we already called meetingManager.leave() to get here
+        break;
+      case MeetingSessionStatusCode.AudioJoinedFromAnotherDevice:
+        console.log('[MeetingManager audioVideoDidStop] Meeting joined from another device');
+        this.meetingStatus = MeetingStatus.JoinedFromAnotherDevice;
+        this.publishMeetingStatus();
+        this.leave();
+        break;
+      // The following status codes are Failures according to MeetingSessionStatus
+      case MeetingSessionStatusCode.AudioAuthenticationRejected:
```

Comment:
I think you can simply check this using `sessionStatus.isFailure()` right?

Reply:
Yup I guess thats better